core/server/worker/src/main/java/alluxio/worker/netty/AbstractWriteHandler.java

```diff
@@ -194,7 +188,9 @@ public void channelRead(ChannelHandlerContext ctx, Object object) throws Excepti
     Protocol.WriteRequest writeRequest = msg.getMessage().asWriteRequest();
     // Only initialize (open the readers) if this is the first packet in the block/file.
     if (writeRequest.getOffset() == 0) {
-      initializeRequest(msg);
+      Preconditions.checkState(mRequest == null);
+      mPosToQueue = 0;
```

Comment:
@apc999 The reason this broke was that the old `initializeRequest` set `mPosToQueue` but the new `reset` sets `mPosToWrite`.